crypto/src/asset/id.rs

```diff
@@ -37,6 +39,24 @@ impl Id {
     }
 }
 
+impl Protobuf<pb::AssetId> for Id {}
+
+impl From<Id> for pb::AssetId {
+    fn from(ai: Id) -> Self {
+        pb::AssetId {
+            asset_id: ai.to_bech32().as_bytes().to_vec(),
+        }
+    }
+}
+
+impl TryFrom<pb::AssetId> for Id {
+    type Error = anyhow::Error;
+
+    fn try_from(value: pb::AssetId) -> Result<Self, Self::Error> {
+        value.asset_id.try_into()
+    }
+}
```

Comment:
It looks like this is making the Protobuf AssetId store the bytes of the bech32 encoding, but since the proto is already a binary format, I think it would be cleaner if it just did `asset_id: ai.0.to_bytes()` on the way out to put the raw bytes of the asset ID in the proto.

To make the proto be Bech32 encoded when using JSON or some other serde serializer, you'd use the `build.rs` to stick the right annotations on the generated proto type, there are some existing examples to copy there.